### PR TITLE
1750: Endpoint specification for verein360

### PR DESCRIPTION
**Pull request by ztefanie**
### Short description
Added documentation we can sent to verein360. As soon as this PR is approved I will send this documentation to Verein360. The deadline for this is end of november.

### Testing
 you can call the endpoint as specified and check if this works as expected.

### Resolved issues

Fixes: #1750

### CODE REVIEW

File: docs/verein360-enpoint.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                "telephone": {"shortText": "0150123456789"}
+              },
+              "name": {
+                "shortText": "Sportverein Augsburg-Nord"
+              }
+            },
+            "payment": false,                         // Must be false
+            "responsibility": {"shortText": "Trainer"},
+            "workSinceDate": {"date": "2020-10-06"},
+            "isAlreadyVerified": true                 // Must be true, so that we can mark the application as verified.
```
Comment (ztefanie) [inline docs/verein360-enpoint.md:73]: Please let me know what we want to do in this case.
We could
1) Send no error message and just accept the applicaiton and not mark it as verified
2) Send an unauthorized error message and not save the application

I think I would prefer 2) to ensure verein360 fixes auth problems and not make the municipalities a lot of work of verifying applications manullay.

File: docs/verein360-enpoint.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+## Introduction
+Verein360/BLSV should be able to send already approved entitlementcard application for their trainers.
+An api token that can be generated from the "Digitale Druckerei" account is required to access the endpoint.
+All details about the endpoint can be found in the [API Documentation](../specs/backend-api.graphql).
+
+## Usage
+
+### Set Authourization Header
+
+Set the HTTP header `Authorization Bearer <api-token>` to access the endpoint.
```
Comment (steffenkleinle) [inline docs/verein360-enpoint.md:12]:
```suggestion
Set the HTTP `Authorization` header to `Bearer <api-token>` to access the endpoint.
```

File: docs/verein360-enpoint.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
+# Verein360 Endpoint Documentation
+
+## Introduction
+Verein360/BLSV should be able to send already approved entitlementcard application for their trainers.
+An api token that can be generated from the "Digitale Druckerei" account is required to access the endpoint.
+All details about the endpoint can be found in the [API Documentation](../specs/backend-api.graphql).
+
+## Usage
+
+### Set Authourization Header
```
Comment (steffenkleinle) [inline docs/verein360-enpoint.md:10]:
```suggestion
### Authorization Header
```

File: docs/verein360-enpoint.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      "givenInformationIsCorrectAndComplete": true,   // Must be true
+      "hasAcceptedEmailUsage": true,                  // true or false
+      "hasAcceptedPrivacyPolicy": true,               // Must be true
+      "wantsDigitalCard": true,                       // Must be true
+      "wantsPhysicalCard": false,                     // Must be false
+      "blueCardEntitlement": {
+        "entitlementType": "WORK_AT_ORGANIZATIONS",   // Must be WORK_AT_ORGANIZATIONS
+        "workAtOrganizationsEntitlement": {
+          "list": {
+            "amountOfWork": 7.5,
```
Comment (steffenkleinle) [inline docs/verein360-enpoint.md:40]: :wrench: Should we add info on the unit? Hours/Week? Hours/Month?

File: docs/verein360-enpoint.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+            "organization": {
+              "address": {
+                "country": {
+                  "shortText": "Germany"
+                },
+                "houseNumber": {
+                  "shortText": "123"
+                },
+                "location": {
+                  "shortText": "Munich"
```
Comment (steffenkleinle) [inline docs/verein360-enpoint.md:50]: :wrench: Same for occurrences below:
```suggestion
                  "shortText": "München"
```

File: docs/verein360-enpoint.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+      "wantsPhysicalCard": false,                     // Must be false
+      "blueCardEntitlement": {
+        "entitlementType": "WORK_AT_ORGANIZATIONS",   // Must be WORK_AT_ORGANIZATIONS
+        "workAtOrganizationsEntitlement": {
+          "list": {
+            "amountOfWork": 7.5,
+            "organization": {
+              "address": {
+                "country": {
+                  "shortText": "Germany"
```
Comment (steffenkleinle) [inline docs/verein360-enpoint.md:44]: :question: Why are these `shortText` attributes needed?
Not sure about this one:
```suggestion
                  "shortText": "Deutschland"
```

File: docs/verein360-enpoint.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+                },
+                "houseNumber": {
+                  "shortText": "123"
+                },
+                "location": {
+                  "shortText": "Munich"
+                },
+                "postalCode": {
+                  "shortText": "80331"
```
Comment (steffenkleinle) [inline docs/verein360-enpoint.md:53]: :thinking: 
```suggestion
                  "shortText": 80331
```

File: docs/verein360-enpoint.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+  "project": "bayern.ehrenamtskarte.app"              // Must be bayern.ehrenamtskarte.app  
+}
+```
+
+
+### Get Region ID for application
+
+#### Get all Regions
+
+The region ID can be obtained by calling the following query:
```
Comment (steffenkleinle) [inline docs/verein360-enpoint.md:124]:
```suggestion
All available regions and their IDs can be obtained by calling the following graphql endpoint:
```

**Review by steffenkleinle — COMMENTED** [review]
I am not really sure of the context of this doc, but perhaps it makes sense to add a little more information on the used types? E.g. that dates need to be ISO8601 date strings.
Depending on the context my comments might be unnecessary.

File: docs/verein360-enpoint.md (hunk 7 above)
Comment (f1sh1918) [reply docs/verein360-enpoint.md:53]: its a string. thats okay

File: docs/verein360-enpoint.md (hunk 6 above)
Comment (f1sh1918) [reply docs/verein360-enpoint.md:44]: if you send an application you can check the actual payload. But good question why this was implemented like this, maybe @michael-markl can give some input

File: docs/verein360-enpoint.md (hunk 1 above)
Comment (f1sh1918) [reply docs/verein360-enpoint.md:73]: But as we discussed this source may also be defined via the purpose of the bearer token or a particular role for verein 360. We may add some meta data for API Tokens in general

**Review by f1sh1918 — APPROVED** [review]
For me this looks good. Just checked the payload.
Two minor things:

1. We should clarify how we get the "source" of application. But as i commented this can also be applied via the api token. I think thats better than provide here an optional string field.
2. I would just add some comments to the dates (format : yyyy-mm-dd f.e.)

File: docs/verein360-enpoint.md [hunk 9]
```diff
@@ -7,9 +7,9 @@ All details about the endpoint can be found in the [API Documentation](../specs/
 
 ## Usage
 
-### Set Authourization Header
+### Authourization Header
```
Comment (steffenkleinle) [inline docs/verein360-enpoint.md:10]: :upside_down_face: Typo
```suggestion
### Authorization Header
```

**Review by steffenkleinle — APPROVED** [review]
LGTM

**Pull request merged**